failing docker tests locally on macOS (case 2)

### Bug description
<!-- Use this section to clearly and concisely describe the bug. -->
With the latest versions of macOS (Catalina 10.15.4) and Docker CE (2.2.0.5), many of the tests fail to run locally. _(This issue was split from #881 after the approach in #882 did not seem to apply to these failures.)_

#### Expected behaviour
<!-- Tell us what you thought would happen. -->
Running the test suite locally on a released version of `repo2docker` should have no errors or failures.

#### Actual behaviour
<!-- Tell us what you actually happens. -->
A number of tests which require running docker fail. There appear to be a number of different failure messages, many appear to be related to failures using `conda` or `pip` to install packages inside the docker container during `self.build()` invocations. Typical message:
```
>                           raise docker.errors.BuildError(l["error"], build_log="")
E                           docker.errors.BuildError: The command '/bin/sh -c apt-get -qq update &&     apt-get -qq install --yes        libapparmo
r1        lsb-release        psmisc        sudo        > /dev/null &&     apt-get -qq purge &&     apt-get -qq clean &&     rm -rf /var/lib/apt/lis
ts/*' returned a non-zero code: 100

repo2docker/app.py:729: BuildError
__________________________________________________________________ test session ___________________________________________________________________

    def test():
        app = make_r2d(args)
        app.initialize()
        if app.run_cmd:
            # verify test, run it
            app.start()
            return
        # no run_cmd given, starting notebook server
        app.run = False
>       app.start()  # This just build the image and does not run it.

tests/conftest.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
repo2docker/app.py:743: in start
    self.build()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```
Step 1 will be to enumerate and categorized those messages.

### How to reproduce
<!-- Use this section to describe the steps that a user would take to experience this bug. -->

1. On a macOS box, checkout a released version of `repo2docker`
2. Setup the development environment.
3. invoke `pytest tests/`
4. observe failures.

### Your personal set up
<!-- Tell us a little about the system you're using. You can see the guidelines for setting up and reporting this information at https://repo2docker.readthedocs.io/en/latest/contributing/contributing.html#setting-up-for-local-development. -->

 - OS: macOS
 - Docker version: CE 2.2.0.5
 - repo2docker version: 0.11.4

--- a/docs/source/contributing/tasks.md
+++ b/docs/source/contributing/tasks.md
@@ -23,6 +23,16 @@ If you want to run a specific test, you can do so with:
 py.test -s tests/<path-to-test>
 ```
 
+### Troubleshooting Tests
+
+Some of the tests have non-python requirements for your development machine. They are:
+
+- `git-lfs` must be installed ([instructions](https://github.com/git-lfs/git-lfs)). It need not be activated -- there is no need to run the `git lfs install` command. It just needs to be available to the test suite. 
+   - If your test failure messages include "`git-lfs filter-process: git-lfs: command not found`", this step should address the problem.
+
+- Minimum Docker Image size of 128GB is required. If you are not running docker on a linux OS, you may need to expand the runtime image size for your installation. See Docker's instructions for [macOS](https://docs.docker.com/docker-for-mac/space/) or [Windows 10](https://docs.docker.com/docker-for-windows/#resources) for more information.
+    - If your test failure messages include "`No space left on device: '/home/...`", this step should address the problem.
+
 ## Update and Freeze BuildPack Dependencies
 
 This section covers the process by which repo2docker defines and updates the

